backend/src/api/open_source_project.ts

```diff
@@ -0,0 +1,263 @@
+ *    tags:
+ *    - Organizations
+ */
+export const del = wrapHandler(async (event) => {
+  const id = event.pathParameters?.orgId;
+  const projectId = event.pathParameters?.projectId;
+
+  if (!id || !projectId ) {
+    return NotFound;
+  }
+
+  if (!isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(id)) return Unauthorized;
+
+  await connectToDatabase();
+
+  try {
+    const organization = await Organization.findOne(id, { relations: ['openSourceProjects'] });
+    if (!organization) {
+      return NotFound;
+    }
+
+    const openSourceProject = organization.openSourceProjects.find(proj => proj.id === projectId);
+    if (!openSourceProject) {
+      return NotFound;
+    }
+
+   // Use QueryBuilder to load the organizations relation for the open source project -- is there an easier way to do this?
+   const openSourceProjectWithOrganizations = await OpenSourceProject.createQueryBuilder('osp')
+   .leftJoinAndSelect('osp.organizations', 'organization')
+   .where('osp.id = :projectId', { projectId })
+   .getOne();
+
+ if (!openSourceProjectWithOrganizations) {
+   return NotFound;
+ }
+
+ const orgIds = openSourceProjectWithOrganizations.organizations.map(org => org.id);
+
+ if (orgIds.length > 1) {
+   // Disconnect the open source project from the organization
+   organization.openSourceProjects = organization.openSourceProjects.filter(proj => proj.id !== projectId);
+   await organization.save();
+
+   return {
+     statusCode: 200,
+     body: JSON.stringify({ message: 'Project disconnected successfully' }),
+   };
+ } else {
+   // Delete the project
+   await openSourceProjectWithOrganizations.remove();
+
+   return {
+     statusCode: 200,
+     body: JSON.stringify({ message: 'Project deleted successfully' }),
+   };
+ }
+
+} catch (error) {
+ return {
+   statusCode: 404,
+   body: JSON.stringify(error)
+ };
+}
+});
+
+
+
+
+/**
+ * @swagger
+ *
+ * /project/grabInfo/{projectId}:
+ *  get:
+ *    description: Get information about a particular open source project.
+ *    parameters:
+ *      - in: path
+ *        name: projectId
+ *        description: Open Source Project ID
+ *    tags:
+ *    - OpenSourceProjects
+ */
+export const getById = wrapHandler(async (event) => {
+  const id = event.pathParameters?.projectId;
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({ error: 'Project ID is required' }),
+    };
+  }
+ 
+  await connectToDatabase();
+
+  const project = await OpenSourceProject.findOne(id, {
+    relations: ['organizations'],
+  });
+
+  if (!project) {
+    return {
+      statusCode: 404,
+      body: JSON.stringify({ error: 'Project not found' }),
+    };
+  }
+ 
+  // retrieve associated orgid(s)
+  const orgIds = project.organizations.map(org => org.id);
+  const userOrgs = getOrgMemberships(event);
+  
+  if (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !userOrgs.some(userId => orgIds.includes(userId))) return Unauthorized;
```

Comment:
```suggestion
  if (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && (new Set(userOrgs)).intersection(new Set(orgIds)).size == 0) return Unauthorized;
```

Small optimization, but comparing as sets should be faster than iterating through all items in both arrays.